From 6aa4547b86701305f4bc3e43d4288a2948e32b29 Mon Sep 17 00:00:00 2001
From: Luwei Zhou <b45643@freescale.com>
Date: Tue, 27 Aug 2013 17:17:28 +0800
Subject: [PATCH 0954/1072] ENGR00276865 hwmon: mma8451: Change warning to
 error

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit 89808feab1a5a7ad1f49e7bbdb4f8f13894036fc

Change the string tranform warning to error.

Signed-off-by: Luwei Zhou <b45643@freescale.com>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 drivers/hwmon/mxc_mma8451.c |   14 ++++++++++----
 1 files changed, 10 insertions(+), 4 deletions(-)

diff --git a/drivers/hwmon/mxc_mma8451.c b/drivers/hwmon/mxc_mma8451.c
index 6a5722e..ddac67d 100644
--- a/drivers/hwmon/mxc_mma8451.c
+++ b/drivers/hwmon/mxc_mma8451.c
@@ -274,8 +274,11 @@ static ssize_t mma8451_enable_store(struct device *dev,
 	u8 val = 0;
 
 	ret = strict_strtoul(buf, 10, &enable);
-	if (ret)
-		dev_warn(dev, "Warning when tranfser string\n");
+	if (ret) {
+		dev_err(dev, "string transform error\n");
+		return ret;
+	}
+
 	mutex_lock(&mma8451_lock);
 	client = mma8451_i2c_client;
 	enable = (enable > 0) ? 1 : 0;
@@ -317,8 +320,11 @@ static ssize_t mma8451_position_store(struct device *dev,
 	unsigned long  position;
 	int ret;
 	ret = strict_strtoul(buf, 10, &position);
-	if (ret)
-		dev_warn(dev, "Warning when tranfser string\n");
+	if (ret) {
+		dev_err(dev, "string transform error\n");
+		return ret;
+	}
+
 	mutex_lock(&mma8451_lock);
 	mma_status.position = (int)position;
 	mutex_unlock(&mma8451_lock);
-- 
1.7.5.4

